src/DssProxyActionsStETH.sol

```diff
@@ -0,0 +1,481 @@
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.6.12;
+
+interface GemLike {
+    function approve(address, uint256) external;
+    function transfer(address, uint256) external;
+    function transferFrom(address, address, uint256) external;
+    function wrap(uint256 ) external returns (uint256);
+    function unwrap(uint256) external returns (uint256);
+    function stETH() external view returns (address);
+    function getWstETHByStETH(uint256) external view returns (uint256);
+}
+
+interface ManagerLike {
+    function ilks(uint256) external view returns (bytes32);
+    function owns(uint256) external view returns (address);
+    function urns(uint256) external view returns (address);
+    function vat() external view returns (address);
+    function open(bytes32, address) external returns (uint256);
+    function give(uint256, address) external;
+    function frob(uint256, int256, int256) external;
+    function flux(uint256, address, uint256) external;
+    function move(uint256, address, uint256) external;
+    function exit(address, uint256, address, uint256) external;
+    function quit(uint256, address) external;
+}
+
+interface VatLike {
+    function can(address, address) external view returns (uint256);
+    function ilks(bytes32) external view returns (uint256, uint256, uint256, uint256, uint256);
+    function dai(address) external view returns (uint256);
+    function urns(bytes32, address) external view returns (uint256, uint256);
+    function frob(bytes32, address, address, address, int256, int256) external;
+    function hope(address) external;
+}
+
+interface GemJoinLike {
+    function dec() external returns (uint256);
+    function gem() external returns (GemLike);
+    function join(address, uint256) external payable;
+    function exit(address, uint256) external;
+}
+
+interface DaiJoinLike {
+    function vat() external returns (VatLike);
+    function dai() external returns (GemLike);
+    function join(address, uint256) external payable;
+    function exit(address, uint256) external;
+}
+
+interface EndLike {
+    function fix(bytes32) external view returns (uint256);
+    function cash(bytes32, uint256) external;
+    function free(bytes32) external;
+    function skim(bytes32, address) external;
+}
+
+interface JugLike {
+    function drip(bytes32) external returns (uint256);
+}
+
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+// WARNING: These functions meant to be used as a a library for a DSProxy. Some are unsafe if you call them directly.
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+contract Common {
+    uint256 constant RAY = 10 ** 27;
+
+    VatLike     immutable public vat;
+    ManagerLike immutable public manager;
```

Comment:
The new proxy actions use this method: https://github.com/makerdao/dss-proxy-actions/blob/dev/src/DssProxyActions.sol (still WIP)
I think the trade off is not bad, it will be quite weird we have a cdp manager change in the near future.

Reply:
@gbalabasquer I didn't fully understand your last sentence. Are you in favor of leaving it as immutbale?

Reply:
Yes, I'd leave it as an immutable. I'm even thinking if other stuff like `daiJoin` should also be (also thinking for the main proxy actions).

Reply:
👍🏻

Reply:
(won't resolve if @gbalabasquer wants other things immutable)